lib/ec2imgutils/ec2imgutils.py

```diff
@@ -41,10 +46,16 @@ def __init__(self, log_level=logging.INFO, log_callback=None):
         except AttributeError:
             self.log_level = self.log.logger.level  # LoggerAdapter
 
+        self.durable_ec2_client = ec2_client
+
     # ---------------------------------------------------------------------
     def _connect(self):
         """Connect to EC2"""
 
+        # Allow dependency injection for easier testing
+        if self.durable_ec2_client:
+            return self.durable_ec2_client
```

Comment:
But the client is not durable by definition. Neither boto3 nor botocore manage the connect. They basically just pass the connection to the API through. This means the connection **will** time out. As such at some point the durable client returned will be useless and broken.

Reply:
Afaik there are ways to self manage sessions and set "TTL". That would provide a way to know when session is expiring and do a refresh. But that falls into a more expanded topic that is beyond the scope of this PR at the moment.